lib/nostr.js

```diff
@@ -66,16 +67,30 @@ export default class Nostr {
 
   /**
    *
-   * @param {Object} param0
+   * @param {Object} args
+   * @param {string} [args.userPreferences] - user given preferences
+   * @param {string} [args.userPreferences.signerType] - signer type to use for signing (nip07, nip46)
+   * @param {string} [args.userPreferences.signer] - signer config to use for signing
    * @param {string} [args.privKey] - private key to use for signing
    * @param {string} [args.nip46token] - NIP-46 token to use for signing
-   * @param {boolean} [args.supportNip07] - whether to use NIP-07 signer if available
+   * @param {boolean} [args.nip07] - whether to use NIP-07 signer if available
    * @returns {NDKPrivateKeySigner | NDKNip46Signer | NDKNip07Signer | null} - a signer instance
    */
-  getSigner ({ privKey, nip46token, supportNip07 = true } = {}) {
+  getSigner ({ userPreferences: { signerType: userSignerTypePreference, signer: userSignerPreference, signerInstanceKey } = {}, privKey, nip46token, nip07, nip46LocalSigner } = {}) {
```

Comment:
`signerInstanceKey` is the [client key pair](https://github.com/nostr-protocol/nips/blob/master/46.md#terminology), which is one of the key parts needed to reestablish the connection for NIP-46 crossposts across user sessions.